services/src/main/java/org/exoplatform/webconferencing/cometd/CometdWebConferencingService.java

```diff
@@ -0,0 +1,851 @@
+
+  public static final String             COMMAND_DELETE                        = "delete";
+
+  public static final String             COMMAND_GET_CALLS_STATE               = "get_calls_state";
+
+  private static final Log               LOG                                   =
+                                             ExoLogger.getLogger(CometdWebConferencingService.class);
+
+  protected final WebConferencingService webConferencing;
+
+  protected final EXoContinuationBayeux  exoBayeux;
+
+  protected final CallService            service;
+
+  protected final OrganizationService    organization;
+
+  protected final SessionProviderService sessionProviders;
+
+  protected final IdentityRegistry       identityRegistry;
+
+  @Service("webconferencing")
+  public class CallService {
+
+    class ChannelContext {
+      final Set<String>      clients = ConcurrentHashMap.newKeySet();
+
+      final UserCallListener listener;
+
+      ChannelContext(UserCallListener listener) {
+        super();
+        this.listener = listener;
+      }
+
+      UserCallListener getListener() {
+        return listener;
+      }
+
+      boolean hasNoClients() {
+        return clients.isEmpty();
+      }
+
+      boolean hasClient(String clientId) {
+        return clients.contains(clientId);
+      }
+
+      boolean removeClient(String clientId) {
+        boolean res = clients.remove(clientId);
+        if (clients.size() == 0) {
+          webConferencing.removeUserCallListener(listener);
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< Removed user call listener for " + listener.getUserId() + ", client:" + clientId);
+          }
+        } else if (res) {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< Removed user call client for " + listener.getUserId() + ", client:" + clientId);
+          }
+        } else {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< User call client was not removed for " + listener.getUserId() + ", client:" + clientId);
+          }
+        }
+        return res;
+      }
+
+      boolean addClient(String clientId) {
+        boolean wasEmpty = clients.size() == 0;
+        boolean res = clients.add(clientId);
+        if (wasEmpty && res) {
+          webConferencing.addUserCallListener(listener);
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< Added user call listener for " + listener.getUserId() + ", client:" + clientId);
+          }
+        } else if (res) {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< Added user call client for " + listener.getUserId() + ", client:" + clientId);
+          }
+        } else {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("<<< User call client was not added for " + listener.getUserId() + ", client:" + clientId);
+          }
+        }
+        return res;
+      }
+    }
+
+    @Deprecated
+    class SessionRemoveListener implements RemoveListener {
```

Comment:
We should remove all the deprecated things IMO. No need to keep them since no previous version exists.